### PR TITLE
feat(linux.rs): Add support for Funtoo

### PR DESCRIPTION
**Pull request by Lazerbeak12345**
## What does this PR do

Adds support for updating [Funtoo Linux](https://www.funtoo.org) systems.

## Standards checklist

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself (I tested it on Funtoo linux ONLY. I'd advise testing it on gentoo as well)
 
## For new steps

- [ ] *Optional:* Topgrade skips this step where needed
- [ ] *Optional:* The `--dry-run` option works with this step
- [ ] *Optional:* The `--yes` option works with this step if it is supported by 
  the underlying command

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

### CODE REVIEW

**Review by SteveLauC — COMMENTED** [review]
HI, thanks for the PR!

I wonder can we have an "os-release" file [there](https://github.com/topgrade-rs/topgrade/tree/main/src/steps/os/os_release), and also a unit test like [this](https://github.com/topgrade-rs/topgrade/blob/07118fa0d25aba573c36a874375caea8c768cd33/src/steps/os/linux.rs#L1183-L1186) :)

File: src/steps/os/linux.rs [hunk 1]
```diff
@@ -452,16 +452,21 @@ fn upgrade_gentoo(ctx: &ExecutionContext) -> Result<()> {
...
-        .execute(sudo)
-        .args(["emerge", "--sync"])
-        .args(
-            ctx.config()
-                .emerge_sync_flags()
-                .map(|s| s.split_whitespace().collect())
-                .unwrap_or_else(|| vec!["-q"]),
-        )
-        .status_checked()?;
+    if let Some(ego) = which("ego") {
```
[the remark this answers is not in the document]
Comment (Lazerbeak12345) [reply src/steps/os/linux.rs:455]: It's Funtoo exclusive.

[the remark this answers is not in the document]
Comment (SteveLauC) [reply src/steps/os/linux.rs:455]: Get it, thanks for the reply.

**Comment by Lazerbeak12345** [comment]
Got those pretty easy.

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**